torch/_inductor/index_propagation.py

```diff
@@ -258,5 +258,10 @@ def indirect_indexing(
         if isinstance(index, IndexPropVar) and index.is_symbolic:
             # If we are turning a indirect indexing into direct, we need to wrap it.
             index = index.value.expr
-            return index + Where(index >= 0, 0, size)
+            result = index + Where(index >= 0, 0, size)
+            var = self.materialize_expr(result, torch.int64)
```

Comment:
This doesn't look right? Why do we need to call materialize_expr at all? Check_bounds should just work with the sympy expression, no?